untitled/test/com/aca/exam/exam2/FileSourceUserRepositoryTest.java

```diff
@@ -0,0 +1,63 @@
+package com.aca.exam.exam2;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Scanner;
+
+class FileSourceUserRepositoryTest {
+    private FileSourceUserRepository testSubject;
+
+    @BeforeEach
+    public void setUp() {
+        testSubject = FileSourceUserRepository.getInstance();
+    }
+
+    @Test
+    public void testGetInstance() {
+        Assertions.assertNotNull(testSubject);
+        FileSourceUserRepository fileSourceUserRepository = FileSourceUserRepository.getInstance();
+        Assertions.assertEquals(fileSourceUserRepository, testSubject);
+    }
+
+    @Test
+    public void testFindByIdWhenIdIsNotFound() {
+        IllegalArgumentException illegalArgumentException = Assertions.assertThrows(IllegalArgumentException.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                testSubject.findById(null);
+            }
+        });
+        Assertions.assertEquals("id can not be null", illegalArgumentException.getMessage());
+    }
+
+    @Test
+    public void testFindByIdWhenIdIsFound() {
+        User user = new User("barracudaassign", "Kaiya", "Dunlap");
+        Assertions.assertTrue(user.equals(testSubject.findById("barracudaassign")));
+    }
+
+    @Test
+    public void testFindAll() {
+        File file = new File("D:\\Users\\user\\IdeaProjects\\aca\\Java_Level_01\\untitled\\main\\\\com\\aca\\exam\\exam2\\User.txt");
```

Comment:
there is no need to duplicate the code in the test method. you should just assert that the findAll size is equal to what you expect. and would be better if you created list of users and asserted that the list are the same